### DISCUSSION TITLE
Import and open sample project

### DISCUSSION THREAD

**Discussion by EOP32** [arkivanov/Decompose#693] [Q&A] [answered]
Hi there! Thank you for this great library. I am pretty new to compose multiplatform and I am trying to use your library in my project. Therefore I am trying to debug your sample app and get a better understanding of how to use the components and how the library works. 

Problem is that after cloning the sample project android studio does not seem to correctly "detect" the android app module. So that I cannot add a run config for android and for example jump to declarations using command + click. 

Is there anything that I need to pay attention to? Or how can I solve it? Thanks in advance.

**Comment by arkivanov** [comment]
Are there any Gradle sync or build errors? Make sure you open the `build.gradle.kts` file from the root folder, and that you created `local.properties` file with Android SDK location in the same root folder.

I use IDEA, but Android Studio should also work.

**Comment by EOP32** [comment]
It actually asks me to link the gradle project.. not very much which steps I need to take here.. 

<img width="1588" alt="image" src="https://github.com/arkivanov/Decompose/assets/62963331/3cb5052e-14a8-4f0f-b835-225d3f07d135">

**Comment by malliaridis** [comment, marked as answer]
Try to open the Decompose project / directory instead of sample, so that Android Studio will be able to load the project correctly. After that, you should see the run configuration on the top with the name "sample.app-android".

This is a common mistake if you are new in Gradle and/or Kotlin Multipaltform. You have opened the sample directory of the project. This however does not include the gradle build files that are located in `Decompose/` (outside of sample). Therefore, it cannot find the "entry point" for gradle and asks you to configure the project. The files `/settings.gradle.kts`, `/build.gradle.kts` are basically the project configuration files, and anything in sub folders are the module configuration files, usually and namely the `build.gradle.kts`.

The `build.gradle.kts` file you marked in the screenshot is the build file of a single module of the project and does not include the necessary information to sync the project / sample (sub)project.

**Comment by EOP32** [comment]
Totally makes sense! Thanks :) It works now
